### PR TITLE
Replace `image` crate jpeg decoder with zune-jpeg

### PR DESCRIPTION
**Pull request by emilk**
### Why
It is much faster.

On my Mac, it takes objectron JPEG decoding from 15 to 10 ms. On the web the difference is smaller: 35ms -> 29ms. See https://github.com/emilk/image-decode-bench for more.

It can also decode directly to RGBA, which means we don't need to spend cycles later padding RGB to RGBA. This takes the entire `objectron` demo `App::update` down from 21 ms to 14 ms.

Sadly, the `gltf` crate still use the `"jpeg"` feature of the `image` crate, so weneed to pay for the compilation and code bloat of two jpeg decoders. The `.wasm` of the web-viewer increases by 23kB (pre-compression) because of this PR.

This PR also adds support for monochrome JPEGs.

### Related
* https://github.com/image-rs/image/issues/1845
* https://github.com/image-rs/image/pull/1876

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

PR Build Summary: https://build.rerun.io/pr/2376

Docs preview: https://rerun.io/preview/1aae8d4/docs
Examples preview: https://rerun.io/preview/1aae8d4/examples

### CODE REVIEW

**Comment by teh-cmc** [comment]
Seeing similar behavior on my _very_ different machine: ~14ms on `main` vs. ~8ms on this branch.

File: crates/re_components/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ ecolor = ["dep:ecolor"]
 glam = ["dep:glam"]
 
 ## Integration with the [`image`](https://crates.io/crates/image/) crate.
```
Comment (teh-cmc) [inline crates/re_components/Cargo.toml:32]: guess that feature description is a bit of a lie now

File: crates/re_components/src/tensor.rs [hunk 2]
```diff
@@ -1196,26 +1191,78 @@ impl DecodedTensor {
...
+        let (w, h) = (w as u64, h as u64);
+
+        if w != expected_width || h != expected_height {
+            return Err(TensorImageLoadError::InvalidMetaData {
+                expected: [expected_height, expected_width, expected_channels].into(),
+                found: [h, w, depth].into(),
+            });
+        }
+
+        assert_eq!(pixels.len() as u64, w * h * depth, "Bug in JPEG decoder");
```
Comment (teh-cmc) [inline crates/re_components/src/tensor.rs:1249]: Still not a good reason to crash the user's app though

**Review by teh-cmc — APPROVED** [review]
:+1: 

Pre-approving but let's get rid of that scary `assert` please :grimacing:

**Comment by emilk** [comment]
Worth noting: this slows down debug builds :(

`zune-jpeg` is 3x slower in debug builds than in release builds, even though we optimize all our dependencies even in debug builds with this:

https://github.com/rerun-io/rerun/blob/4acfcbd0d62da66f331d646a66bce50d038def02/Cargo.toml#L122-L124

This is strange

**Pull request merged**